### DISCUSSION TITLE
Python-SocketIO >= 5.7.0 rejecting connections to namespaces created via external processes using Redis message broker

### DISCUSSION THREAD

**Discussion by Ghidra3x7** [miguelgrinberg/python-socketio#975] [Q&A] [answered]
**Describe the bug**
An external process is emitting events to a namespaces using Redis as the broker. When using python-socketio <= 5.6, Gunicorn, Flask-SocketIO==5.2.0 and python-engineio==4.3.3, the socketio client can connect to the namespace without a problem. When updating to python-socketio >= 5.7, the client does not connect. I believe (but not certain), this is due to change "Server refuses connections on unknown namespaces (#822)." I suspect that when an external process emits an event to a namespace using Redis as the message queue, the namespace is not "registered" and the server then refuses connections to the "unknown" namespace.

**To Reproduce**
Use an external process (and Flask-SocketIO) to emit an event to a namespace using Redis as the message broker. A web server (e.g. Gunicorn) loads Flask-SocketIO connects to the Redis broker. A client (using latest socketio 4.5+) tries to connect to the namespace; connection is refused. 

**Expected behavior**
Python-SocketIO <= 5.6 works as expected - I can emit events via an external process to a namespace using Flask-SocketIO and Redis message broker; the client can connect to the namespace without a problem. Starting with 5.7+ , the connection is refused.

**Logs**

Not available at the moment.

**Comment by miguelgrinberg** [comment, marked as answer]
Version 5.7. introduces a change with regards to accepting connections. Because of an unintended omission, previous versions accepted connections from any namespaces. In 5.7 (practically 5.7.1) by default only connections from known namespaces are accepted. Known namespaces are the default namespace `/` and any namespaces for which there are event handlers. If you want to accept connections from other namespaces you have two options:

- Add a `connect` handler for any of the namespaces that you want to accept
- Use the `namespaces` argument to provide the list of namespaces you'd like to accept. If you pass `namespaces='*'` all namespaces will be accepted, which is the unintended behavior of 5.6 and older versions.

**Reply by Ghidra3x7** [reply]
Ah got it, thank you for the quick response! Much appreciated.
